apps/openmw/mwmechanics/spells.cpp

```diff
@@ -166,101 +180,78 @@ namespace MWMechanics
         return false;
     }
 
-    bool Spells::hasCommonDisease() const
+    bool Spells::hasDisease(const ESM::Spell::SpellType type) const
     {
-        for (TIterator iter = mSpells.begin(); iter!=mSpells.end(); ++iter)
+        for (const auto& iter : mSpells)
         {
-            const ESM::Spell *spell = iter->first;
-            if (spell->mData.mType == ESM::Spell::ST_Disease)
+            const ESM::Spell *spell = iter.first;
+            if (spell->mData.mType == type)
                 return true;
         }
 
         return false;
     }
 
-    bool Spells::hasBlightDisease() const
+    bool Spells::hasCommonDisease() const
     {
-        for (TIterator iter = mSpells.begin(); iter!=mSpells.end(); ++iter)
-        {
-            const ESM::Spell *spell = iter->first;
-            if (spell->mData.mType == ESM::Spell::ST_Blight)
-                return true;
-        }
+        return hasDisease(ESM::Spell::ST_Disease);
+    }
 
-        return false;
+    bool Spells::hasBlightDisease() const
+    {
+        return hasDisease(ESM::Spell::ST_Blight);
     }
 
-    void Spells::purgeCommonDisease()
+    void Spells::purge(const std::function<bool(const ESM::Spell*)>& filter)
```

Comment:
Maybe a typedef would be better here.

Reply:
For the template  argument or for the entire thing?

Honestly, not a big fan of typedefs.

Reply:
For the entire argument.

Reply:
How's this?

Reply:
Looks much better.